### PR TITLE
Add Sonar Cloud to list of Code Analysis Tools

### PR DESCRIPTION
**Pull request by matt1as**

### CODE REVIEW

**Review by xtough — CHANGES_REQUESTED** [review]
@larshp FYI, for some reason I cannot add you as reviewer.

@matt1as are you affiliated to Sonarsource? It sounds a bit like marketing, which makes me feel a bit uneasy. See also my inline comments on typos.

File: abap-code-review/src/tools.adoc [hunk 1]
```diff
@@ -120,3 +120,9 @@ pipeline -> pipeline: 4: abaplint
 === abap-openapi-client
 
 External check services often have an link:https://swagger.io/docs/specification/about/[OpenAPI] definition. Unfortunately, there is no standard way of consuming such OpenAPI services in ABAP yet. An emerging open source OpenAPI client for ABAP can be found at https://github.com/abap-openapi/abap-openapi-client
+
+=== SonarCloud
+
+link:https://sonarcloud.io/[SonarCloud] is a product for Continuous Code Quality online, totally free for open-source projects. It supports multiple languages, Abap serialized by AbapGIT being one of them. There are about 100 different rules, with examples and explainations, that can be turned on and off depending on your needs. SonarCloud can be used to seamelessly integrate Abap Code together with checks for other languages that a project might use.
```
[the remark this answers is not in the document]
Comment (matt1as) [reply abap-code-review/src/tools.adoc:126]: I'm not affiliated with SonarSource, we are a paying customers that use SonarCloud for our C#, JS and ABAP code review (together with ATC). We find that the two tools complement each other and catches different kinds of errors/mistakes. I did not know it existed until 2 years ago, and I think there are more ABAP-shops that would benefit from using it.

Please suggest how I could make it less "marketing".

**Comment by xtough** [comment]
Thanks for explaining, @matt1as . And first of all, thanks for your contribution! 👍 

I had a brief conversation with @larshp on this PR. We felt that we should make clear in the tools chapter which kind of tools we are discussing. There are basically 3 types:
1. provided by SAP -- usually part of your support contract with SAP
2. Open source -- Mostly projects from the ecosystem around dotabap.org , best effort support, take it or leave it
3. Third party -- subject to individual licenses (like GitHub, GitLab, and SonarSource for that matter)

The tools chapter is currently structured along tool purposes as a different dimension. Nevertheless, I would prefer to move this section to a "4.5.3 Third party tools" instead of the currently too prominent "4.7 SonarCloud". 

In other words: Move it up between ``==== abapLint`` and ``=== abap-openapi-client`` and increase its title indentation by one ``=``

File: abap-code-review/src/tools.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (larshp) [reply abap-code-review/src/tools.adoc]: as a customer, and vendors basing products on top of opensource, consider choosing vendors which actively support open source

of cause open source is open, but its also open for customers to choose vendors

some people have spent a lot of time sharing, just like this document is shared openly

in my opinion, it's not helping to add a list of commercial only 3rd party vendors

[the remark this answers is not in the document]
Comment (matt1as) [reply abap-code-review/src/tools.adoc]: SonarSource is actually a pretty good open source citizen; most of their products are open source under LGPL, but unfortunately the ABAP check is not one of them. 

They live on github, [https://github.com/SonarSource]

File: abap-code-review/src/tools.adoc [hunk 2]
```diff
@@ -117,6 +117,12 @@ pipeline -> pipeline: 4: abaplint
 @enduml
 ....
 
+==== SonarCloud
+
+link:https://sonarcloud.io/[SonarCloud] is a product for Continuous Code Quality online,  free for open-source projects. It supports multiple languages, ABAP serialized by abapGit being one of them. There are about 100 different rules, with examples and explanations, that can be turned on and off depending on your needs. SonarCloud can be used to seamlessly integrate ABAP-code together with checks for other languages that a project might use.
+
+SonarCloud can be added by using their link:https://github.com/apps/sonarcloud[GitHub App]
+
```
Comment (xtough) [inline abap-code-review/src/tools.adoc:125]:
```suggestion
==== Third party checks

There are other tools from third party vendors that can be also used for ABAP code analysis. 

For instance, link:https://www.sonarsource.com/abap/[SonarSource ABAP] can be used to check ABAP source serialized by abapGit. 
Their link:https://github.com/apps/sonarcloud[App] can be used to integrate it with GitHub-based CI pipelines.
```

**Review by xtough — CHANGES_REQUESTED** [review]
I would suggest to make this even less prominent. So far we don't use Sonarsource in our example or anywhere else in the dotabap.org ecosystem and Sonarsource hasn't contributed anything yet either. It is OK to mention it for academic completeness but there is no need to advertise third party commercial offerings. The section title even leaves room for others...

File: abap-code-review/src/tools.adoc [hunk 3]
```diff
@@ -117,6 +117,12 @@ pipeline -> pipeline: 4: abaplint
 @enduml
 ....
 
+==== Third party checks
+
+There are other tools from third party vendors that can be also used for ABAP code analysis. 
+
+For instance, link:https://www.sonarsource.com/abap/[SonarSource ABAP] can be used to check ABAP source serialized by abapGit. 
+Their link:https://github.com/apps/sonarcloud[App] can be used to integrate it with GitHub-based CI pipelines.
 === abap-openapi-client
```
Comment (xtough) [inline abap-code-review/src/tools.adoc:126]:
```suggestion

=== abap-openapi-client
```
missing line feed to keep formatting intact

**Review by xtough — CHANGES_REQUESTED** [review]
line feed missing, 
sorry, my fault

**Review by xtough — APPROVED** [review]
LGTM.
@larshp any objections?

**Comment by larshp** [comment]
looks good, good to merge 👍
